packages/schematics/src/app/rules.ts

```diff
@@ -0,0 +1,207 @@
+} from '@schematics/angular/utility/json-utils';
+import { getProject } from '@schematics/angular/utility/project';
+import { getProjectTargets } from '@schematics/angular/utility/project-targets';
+import { of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+import { getLatestNodeVersion, NpmRegistryPackage } from './npmjs-utils';
+import { Schema } from './schema';
+
+export function validateOptions(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    if (!options.project || options.project === '') {
+      throw new SchematicsException('You need to enter project name');
+    }
+
+    const workspace = getWorkspace(tree);
+    const project = getProject(workspace, options.project);
+
+    if (!project) {
+      throw new SchematicsException(`Could not find project: '${options.project}'`);
+    }
+
+    return tree;
+  };
+}
+
+export function addPackageJsonDependencies(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    return of('@augury/core', '@augury/performance-profiler-plugin').pipe(
+      concatMap(name => getLatestNodeVersion(name)),
+      map((npmRegistryPackage: NpmRegistryPackage) => {
+        const nodeDependency: NodeDependency = {
+          type: NodeDependencyType.Dev,
+          name: npmRegistryPackage.name,
+          version: npmRegistryPackage.version,
+          overwrite: false,
+        };
+        addPackageJsonDependency(tree, nodeDependency);
+        context.logger.info(`✅️ Added ${nodeDependency.name} v.${nodeDependency.version}`);
+        return tree;
+      }),
+    );
+  };
+}
+
+export function installDependencies(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.addTask(new NodePackageInstallTask());
+    context.logger.info('✅️ Dependencies installed');
+    return tree;
+  };
+}
+
+export function addAuguryMain(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProject(workspace, options.project);
+
+    if (!project || !project.sourceRoot) {
+      throw new SchematicsException(`Project root could not be found`);
+    }
+
+    const sourceDir = normalize(project.sourceRoot);
+    const templSource = apply(url('./files'), [
+      template({
+        ...options,
+      }),
+      move(sourceDir),
+    ]);
+
+    context.logger.info(`✅️ Adding main.augury.ts...`);
+
+    return branchAndMerge(mergeWith(templSource))(tree, context);
+  };
+}
+
+export function addBuildConfiguration(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProject(workspace, options.project);
+    const projectTargets = getProjectTargets(project);
+
+    if (
+      !projectTargets.build ||
+      !projectTargets.build.configurations ||
+      !projectTargets.serve ||
+      !projectTargets.serve.configurations
+    ) {
```

Comment:
maybe refactor to `areProjectTargetsNotFound()`? When if statements get larger than 1 line, I tend to put them in methids/functions to simplify and document the intent.